R/redist.R

```diff
@@ -707,389 +798,267 @@ redist.combine <- function(savename, nsims, nloop, nthin, nunits, temper = 0
 #' nsims = 10000)
 #' }
 #' @export
-redist.mcmc <- function(adjobj, popvec, nsims, ndists = NULL, initcds = NULL,
+redist.mcmc <- function(algorithm = 'mcmc', adjobj, popvec, nsims, ndists = NULL, initcds = NULL,
                         loopscompleted = 0, nloop = 1, nthin = 1, eprob = 0.05,
                         lambda = 0, popcons = NULL, grouppopvec = NULL,
-                        ssdmat = NULL, beta = 0, temper = "none",
-                        constraint = "none",
+                        areasvec = NULL, countymembership = NULL,
+                        borderlength_mat = NULL, ssdmat = NULL, temper = FALSE,
+                        constraint = NULL, constraintweights = NULL,
+                        compactness_metric = "fryer-holden",
                         betaseq = "powerlaw", betaseqlength = 10,
                         betaweights = NULL, 
                         adjswaps = TRUE, rngseed = NULL, maxiterrsg = 5000,
                         adapt_lambda = FALSE, adapt_eprob = FALSE,
                         contiguitymap = "rooks", exact_mh = FALSE, savename = NULL,
-                        verbose = TRUE
-                        ){
-
-    if(verbose){
-        ## Initialize ##
-        divider <- c(paste(rep("=", 20), sep = "", collapse = ""), "\n")
-        
-        cat("\n", append = TRUE)
-        cat(divider, append = TRUE)
-        cat("redist.mcmc(): Automated Redistricting Simulation Using
-         Markov Chain Monte Carlo\n\n", append = TRUE)
-    }
+                        verbose = TRUE,
+                        num_hot_steps = 40000, num_annealing_steps = 60000,

```

Comment:
This does require a bit of C++ code from the `dev` branch, at https://github.com/kosukeimai/redist/blob/dev/src/sw_mh_alg.cpp#L131, https://github.com/kosukeimai/redist/blob/dev/src/sw_mh_alg.cpp#L403, and a few other places if you do CTRL-F "annealing". You should try folding those in to the C++ function in the master branch.